src/utils/files.ts

```diff
@@ -46,10 +46,16 @@ interface TypeScore {
   total: number;
 }
 
-async function csvFileTypeRecommendations(file: File): Promise<Map<string, CSVColumnType>> {
-  const parsePromise: Promise<Map<string, CSVColumnType>> = new Promise((resolve) => {
+interface Recommendation {
+  typeRecs: Map<string, CSVColumnType>;
+  rowSample: Array<{}>;
+}
+
+async function csvFileTypeRecommendations(file: File): Promise<Recommendation> {
```

Comment:
If we're going to be returning multiple things from here, I feel like we should rename these things. The interface could be something like `CSVAnalysisResult`, and the function something like `AnalyzeCSV`. I'm not sure if those are the best names, but we're doing two different things here now, and I think it'd be good to reflect that in the function/type names.

Reply:
Good idea; check out the names I chose: d7ef1fbd31555f78e04484abcb544b653aa5ea4f.